src/identity/actions/thunks/index.ts

```diff
@@ -0,0 +1,121 @@
+import {GetState} from 'src/types'
+import {Dispatch} from 'react'
+
+// Actions
+import {
+  setQuartzMe,
+  setQuartzMeStatus,
+  Actions as MeActions,
+} from 'src/me/actions/creators'
+import {
+  setCurrentOrgDetails,
+  setCurrentBillingProvider,
+  setQuartzIdentity,
+  setQuartzIdentityStatus,
+  Actions as IdentityActions,
+} from 'src/identity/actions/creators'
+
+// Types
+import {RemoteDataState} from 'src/types'
+enum RetrievalError {
+  accountIdentity = 'Unable to retrieve account ID from user identity.',
+  orgIdentity = 'Unable to retrieve organization ID from user identity.',
+}
+
+// Utilities
+import {
+  convertIdentityToMe,
+  fetchQuartzIdentity,
+  fetchAccountDetails,
+  fetchOrgDetails,
+} from 'src/identity/apis/auth'
+
+// Retrieves user's quartz identity from /quartz/identity, and stores it in state.identity.
+export const getQuartzIdentityThunk = () => async (
+  dispatch: Dispatch<IdentityActions | MeActions>
+) => {
+  try {
+    dispatch(setQuartzIdentityStatus(RemoteDataState.Loading))
+
+    const quartzIdentity = await fetchQuartzIdentity()
+
+    dispatch(setQuartzIdentity(quartzIdentity))
+    dispatch(setQuartzIdentityStatus(RemoteDataState.Done))
+
+    const legacyMe = convertIdentityToMe(quartzIdentity)
+    // Remove lines below once quartzMe is deprecated.
+    dispatch(setQuartzMe(legacyMe, RemoteDataState.Done))
+    dispatch(setQuartzMeStatus(RemoteDataState.Done))
+  } catch (error) {
+    dispatch(setQuartzIdentityStatus(RemoteDataState.Error))
+    // Remove line below once quartzMe is deprecated.
+    dispatch(setQuartzMeStatus(RemoteDataState.Error))
+    console.error(error)
+  }
+}
+
+// Retrieves current account's billingProvider, and stores it in state.identity.account.
+export const getBillingProviderThunk = () => async (
+  dispatch: Dispatch<IdentityActions | MeActions>,
+  getState: GetState
+) => {
+  try {
+    dispatch(setQuartzIdentityStatus(RemoteDataState.Loading))
+
+    const initialState = getState()
+    const accountId = initialState?.identity?.account?.id
+
+    if (typeof accountId !== 'string' && typeof accountId !== 'number') {
+      throw new Error(RetrievalError.accountIdentity)
+    }
```

Comment:
in what situation would `accountId` not be a number or string? What case is this guarding against that the `try/catch` block won't guard against?

Reply:
For both the `billing` (i.e. `account`) and `org` thunks, you're right we shouldn't need this. For `account` the `accountId` is guaranteed to be a `number`  and for `org` the type is guaranteed to be a string. (And it's set to those types by default in the reducer too.)